File: JavaScript/Bits/longueurBit.js
/**
 * @description Renvoie le nombre de bits utilisés dans la représentation binaire d'un nombre entier.
 * @param {number} int - Le nombre entier dont on veut connaître la longueur de sa représentation binaire.
 * @return {number} Nombre de bits utilisés.
 */
const longueurBit = (int) => {
  let compteur = 0;
  do compteur++;
  while (1 << compteur <= int);
  return compteur;
};

// Test avec 1, 8 et 512
console.log("Nombre de bits pour 1: ", longueurBit(1)); // 1
console.log("Nombre de bits pour 8: ", longueurBit(8)); // 4
console.log("Nombre de bits pour 512: ", longueurBit(512)); // 10

File: Python/Bits/longueur_bit.py
def longueur_bit(n: int) -> int:
    """
    Description:
        Renvoie le nombre de bits utilisés dans la représentation binaire d'un nombre entier.
    
    Paramètres:
        n: {int} -- Le nombre entier dont on veut connaître la longueur de sa représentation binaire.

    Retourne:
        {int} -- Le nombre de bits utilisés dans la représentation binaire du nombre.
    
    Exemple:
        >>> longueur_bit(1)
        1

        >>> longueur_bit(8)
        4

        >>> longueur_bit(512)
        10

        1 => 1, 8 => 1000, 512 => 1000000000 (décimal => binaire)
    """
    nbBit = 0
    while n > 0:
        nbBit += 1
        n = n >> 1
    return nbBit